app/controllers/users/phone_setup_controller.rb

```diff
@@ -135,5 +137,39 @@ def new_phone_form_params
         :recaptcha_mock_score,
       )
     end
+
+    def check_phone_submission_limit
+      if exceeded_phone_submission_limit?
+        return handle_too_many_phone_submissions
+      end
+      phone_submission_rate_limiter.increment
+      if exceeded_phone_submission_limit?
+        return handle_too_many_phone_submissions
+      end
```

Comment:
Do we need to check the limit twice? Can't we do it once, either before or after the increment?

Reply:
Removed this by reverting course. This pattern exists in https://github.com/18F/identity-idp/blob/45648039e7c3003ad6ebd385d1d376117c3132c7/app/controllers/users/two_factor_authentication_controller.rb#L178-L190